crates/frontend/src/components/type_template_form/mod.rs

```diff
@@ -0,0 +1,119 @@
+        let payload = json!({
+            "type_name": type_name,
+            "type_schema": type_schema
+        });
+        let handle_submit_clone = handle_submit.clone();
+        spawn_local({
+            let handle_submit = handle_submit_clone;
+            async move {
+                let result = create_update_type(tenant_rs.get(), payload.clone()).await;
+                match result {
+                    Ok(_) => {
+                        handle_submit();
+                    }
+                    Err(e) => {
+                        set_error_message.set(e);
+                    }
+                }
+            }
+        });
+    };
+    view! {
+        <form class="form-control w-full space-y-4 bg-white text-gray-700 font-mono">
+            <div class="form-control">
+                <label class="label">
+                    <span class="label-text">Type Name</span>
+                </label>
+                <input
+                    disabled=edit
+                    type="text"
+                    placeholder="Type name"
+                    name="type_name"
+                    id="type_name"
+                    class="input input-bordered w-full max-w-md"
+                    value=move || type_name_rs.get()
+                    on:change=move |ev| {
+                        let value = event_target_value(&ev);
+                        type_name_ws.set(value);
+                    }
+                />
+
+            </div>
+
+            <div class="divider"></div>
+            <div class="form-control">
+                <label class="label">
+                    <span class="label-text">Type Schema</span>
+                </label>
+                <textarea
+                    type="text"
+                    class="input input-bordered shadow-md"
+                    name="type_schema"
+                    id="type_schema"
+                    style="min-height: 150px"
+                    placeholder="type schema"
+                    on:change=move |ev| {
+                        let value = event_target_value(&ev);
+                        match from_str::<Value>(&value) {
+                            Ok(test_val) => {
+                                type_schema_ws.set(test_val);
+                                set_error_message.set("".to_string());
+                            }
+                            Err(err) => {
+                                set_error_message.set(err.to_string());
+                            }
+                        };
+                    }
+                >
+
+                    {move || format!("{}", type_schema_rs.get())}
+                </textarea>
+            </div>
+
+            <div class="form-control grid w-full justify-end">
+                <Button
+                    class="pl-[70px] pr-[70px]".to_string()
+                    text="Submit".to_string()
+                    on_click=on_submit
+                />
+            </div>
+
+            {
```

Comment:
You can remove the `code block` here and `view!` too

Reply:
@Datron , can you address this comment?

Reply:
@Datron , can you address this comment?

Reply:
Done